### ISSUE TITLE
Bug when saving manually drawn labels layer

### ISSUE THREAD

**Issue by haesleinhuepf** [napari/napari#2927]
## 🐛 Bug

Hi all,

I was just trying to save manually drawn labels layer and receive an error. It appears that if the folder/file history is empty, this error happens.

## To Reproduce

* Start napari
* Drag & drop an image on napari
* Create a new labels layer
* Draw something
* Click on File > Save selected layers
This error pops up:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
c:\users\rober\miniconda3\envs\bio9\lib\site-packages\napari\_qt\qt_main_window.py in ()
    485         save_selected_layers.setStatusTip(trans._('Save selected layers'))
    486         save_selected_layers.triggered.connect(
--> 487             lambda: self.qt_viewer._save_layers_dialog(selected=True)
        global self.qt_viewer._save_layers_dialog = undefined
        global selected = undefined
    488         )
    489 

c:\users\rober\miniconda3\envs\bio9\lib\site-packages\napari\_qt\qt_viewer.py in _save_layers_dialog(self=, selected=True)
    531             parent=self,
    532             caption=trans._('Save {msg} layers', msg=msg),
--> 533             directory=hist[0],  # home dir by default,
        global directory = undefined
        hist = []
    534             filter=ext_str,
    535         )

IndexError: list index out of range
```

## Expected behavior

It should be checked if the `hist` array has length > 0.

## Environment

```
napari: 0.4.9
Platform: Windows-10-10.0.19041-SP0
Python: 3.8.5 (default, Sep 3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.0
SciPy: 1.7.0
Dask: 2021.06.2
VisPy: 0.6.6

OpenGL:
- GL version: 4.6.14756 Compatibility Profile Context 20.40.44.02 27.20.14044.2001
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- clEsperanto: 0.9.3
- console: 0.0.3
- scikit-image: 0.4.9
- svg: 0.1.5
```
 - Any other relevant information:

**Comment by jni** [comment]
Was this not fixed in #2876 which fixed #2867 which appears to be the same issue? 🤔 @haesleinhuepf can you check whether this error occurs on master rather than 0.4.9?

**Comment by haesleinhuepf** [comment]
Ha, right! In 0.4.11 it works. Thanks for the headsup!
